flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyConfig.java

```diff
@@ -38,53 +36,6 @@
 
 	private static final Logger LOG = LoggerFactory.getLogger(NettyConfig.class);
 
-	// - Config keys ----------------------------------------------------------
-
-	public static final ConfigOption<Integer> NUM_ARENAS = ConfigOptions
```

Comment:
I'm not sure what the benefit of moving these is. The `NetworkEnvironment` _itself_ isn't even using them (and I supposed never will since they are details of the netty implementation).
I'd keep them here to prevent `NetworkEnvironmentOptions` from becoming a hodge-podge of options for various shuffle-service implementations as well as generic ones.

This is in a way quite similar to how we handle YARN/Mesos/etc.; the `ResourceManagerOptions` contain generic options for all container environments, but we still have distinct `*Options` classes for each.

Reply:
`NetworkEnvironment` relies on `NetworkEnvironmentConfiguration` which is basically a Java parsed version of `NetworkEnvironmentOptions`. `NettyConfig` is part of `NetworkEnvironmentConfiguration`, also parsed and intended for `NetworkEnvironment`. It is all about netty used for shuffling. We also consider netty shuffle implementation to be the default one as it is basically now. I am not sure about splitting its config.  Other shuffle services might reuse these options, might not. Their additional options will be separate anyways, either own class or even outside of Flink.

Reply:
Thanks for sharing the experience in `ResourceManagerOptions` @zentol !
The `NettyConfig` is actually used in current `NetworkEnvironment` which should start `NettyClient` and `NettyServer` based on `NettyConfig`. So all the options in `NetworkEnvironmentOptions` are used by `NetworkEnvironment` atm. But TBH I am also not sure the other shuffle services would reuse it or not, and it might rely on which kind of shuffle service we want to implement future. So the current `NetworkEnvironmentOptions` might be a special one for the default implementation or a general one for some implementations.